canonical-web-and-design/dotrun
# yarn-only: Restructure to remove Poetry support

Since we've decided not to use Poetry for now, as it's too unstable, I'm going to start with scaling back the project to just support basic Yarn dependencies, and then build it up from there.

QA
--

Install the snap:

``` bash
snapcraft
snap install --dangerous dotrun_*.snap
```

Try it out on ubuntu.com:

``` bash
cd ../ubuntu.com
dotrun build
```

This should successfully install dependencies, and build the CSS. It will create a file called `.dotrun.json` for storing the state of dependencies.

Running `dotrun build` should run the build again without needing to install yarn dependencies a second time.

Running `dotrun clean` should delete `.dotrun.json`, `node_modules`, and run `yarn run clean`.

Note, other commands like `dotrun serve` won't work 'cos at the moment there's no support for Python. That's coming in a future PR.
